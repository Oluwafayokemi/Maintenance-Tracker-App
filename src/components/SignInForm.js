import React from 'react';
import '../styles/App.scss';
import history from '../util/history';

export default class SignInForm extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.logInUserRequest(this.state)
      .then((response) => {
        console.log(response.user.user.isAdmin)
        if (response.user.user.isAdmin) {
          history.push('/admin');
        } else if (!response.user.user.isAdmin) {
          history.push('/user');
        }
      })
      .catch((err) => {
        err;
      });
  }

  render() {
    return (
      <React.Fragment>
        <div className="col-6 flex-container">
          <div className="nav">
            <nav>
              <ul className="rform">
                <form id="signin-form" className="one" onSubmit={this.handleSubmit}>
                  <li className="hide2">Sign In</li>
                  <li>
                    <div className="email">
                      <label htmlFor="email">
                        <strong>Email</strong>
                      </label>
                      <input
                        type="email"
                        placeholder="Enter Email"
                        name="email"
                        value={this.state.email}
                        onChange={this.handleChange}
                        required
                      />
                    </div>
                  </li>
                  <li>
                    <div className="psw">
                      <label htmlFor="password">
                        <strong>Password</strong>
                      </label>
                      <input
                        type="password"
                        placeholder="Enter Password"
                        name="password"
                        value={this.state.password}
                        onChange={this.handleChange}
                        required
                      />
                    </div>
                  </li>
                  <button type="submit" value="submit">
                    Login
                  </button>
                </form>
              </ul>
            </nav>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

