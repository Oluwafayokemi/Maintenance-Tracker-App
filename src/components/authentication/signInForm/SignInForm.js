import React from 'react';
import history from '../../../util/history';

export default class SignInForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  handleSubmit = async (event) => {
    event.preventDefault();
    const { logInUserRequest } = await this.props;
    const loginbtn = window.document.querySelector('#login-btn');
    loginbtn.textContent = 'loading...';
    const response = await logInUserRequest(this.state);
    try {
      if (response.user.isAdmin) {
       return history.push('/admin');
      } else if (!response.user.isAdmin) {
        return history.push('/user');
      }
    }
    catch (err) {
      err;
    };
  }

  render() {
    return (
      <React.Fragment>
        <div className="col-6 flex-container">
          <div className="nav">
            <nav>
              <ul className="rform">
                <form id="signin-form" className="one" onSubmit={this.handleSubmit}>
                  <li className="hide2">Sign In</li>
                  <li>
                    <div className="email">
                      <label htmlFor="email">
                        <strong>Email</strong>
                      </label>
                      <input
                        type="email"
                        placeholder="Enter Email"
                        name="email"
                        value={this.state.email}
                        onChange={this.handleChange}
                        required
                      />
                    </div>
                  </li>
                  <li>
                    <div className="psw">
                      <label htmlFor="password">
                        <strong>Password</strong>
                      </label>
                      <input
                        type="password"
                        placeholder="Enter Password"
                        name="password"
                        value={this.state.password}
                        onChange={this.handleChange}
                        required
                      />
                    </div>
                  </li>
                  <button type="submit" value="submit" id="login-btn">
                    Login
                  </button>
                </form>
              </ul>
            </nav>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

