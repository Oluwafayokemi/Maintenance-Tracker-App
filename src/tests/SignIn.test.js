import React from 'react';
import { shallow } from 'enzyme';
import SignInForm from '../components/authentication/signInForm/SignInForm';

const mockEvent = { target: {}, preventDefault: () => jest.fn() }

describe('Tests SignInForm component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<SignInForm />);
    wrapper.find('[name="email"]').simulate('change', { target: { value: 'test@email.com' } });
    wrapper.find('[name="password"]').simulate('change', { target: { value: 'Password' } });
    wrapper.find('form').simulate('submit', mockEvent);
    expect(wrapper).toMatchSnapshot();
  });
});
