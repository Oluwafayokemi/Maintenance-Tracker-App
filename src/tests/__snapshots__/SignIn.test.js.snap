// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests SignInForm component should render without errors 1`] = `
<Fragment>
  <div
    className="col-6 flex-container"
  >
    <div
      className="nav"
    >
      <nav>
        <ul
          className="rform"
        >
          <form
            className="one"
            id="signin-form"
            onSubmit={[Function]}
          >
            <li
              className="hide2"
            >
              Sign In
            </li>
            <li>
              <div
                className="email"
              >
                <label
                  htmlFor="email"
                >
                  <strong>
                    Email
                  </strong>
                </label>
                <input
                  name="email"
                  onChange={[Function]}
                  placeholder="Enter Email"
                  required={true}
                  type="email"
                  value=""
                />
              </div>
            </li>
            <li>
              <div
                className="psw"
              >
                <label
                  htmlFor="password"
                >
                  <strong>
                    Password
                  </strong>
                </label>
                <input
                  name="password"
                  onChange={[Function]}
                  placeholder="Enter Password"
                  required={true}
                  type="password"
                  value=""
                />
              </div>
            </li>
            <button
              id="login-btn"
              type="submit"
              value="submit"
            >
              Login
            </button>
          </form>
        </ul>
      </nav>
    </div>
  </div>
</Fragment>
`;
